### PR TITLE
Reverse complement negative strand reads in fastq output

### LINKED ISSUE #736: adam2fastq does not reverse complement negative strand reads
Pointed out by @timodonnell 

I've got a fix and a few tests coming for this shortly

### PR DESCRIPTION
**Pull request by arahuja**
Fixes #736 

This does slightly differ from Picard.  `AlignmentRecordConverter` currently ignores the negative strand flag is the read is unmapped.  That would result in the read *not* being reverse complemented when output.  I've left as is and only compared with Picard on mapped reads.

This then will leave unmapped reads unaltered on output (which seems to be the correct thing to do) but I'm not sure why the aligner would mark it as negative if it was unmapped.  I.E. if the aligner reverse complements but then removes the mapping (sets the `mapq` to 0) perhaps we should look at the flag?

### CODE REVIEW

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/ADAM2FastqSuite.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package org.bdgenomics.adam.cli
```
Comment (fnothaft) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/ADAM2FastqSuite.scala]: License header.

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/ADAM2FastqSuite.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.io.File
```
Comment (fnothaft) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/ADAM2FastqSuite.scala]: Nit: Can you move this to the end of the imports? We're trying to keep imports alphabetized.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala [hunk 3]
```diff
@@ -18,7 +18,7 @@
 package org.bdgenomics.adam.converters
 
 import htsjdk.samtools.{ SAMFileHeader, SAMRecord }
-import org.bdgenomics.adam.models.{ RecordGroupDictionary, SAMFileHeaderWritable, SequenceDictionary }
+import org.bdgenomics.adam.models.{ Alphabet, RecordGroupDictionary, SAMFileHeaderWritable, SequenceDictionary }
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala]: Maybe just make this `import org.bdgenomics.adam.models._`?

File: adam-cli/src/test/scala/org/bdgenomics/adam/cli/ADAM2FastqSuite.scala (hunk 2 above)
Comment (fnothaft) [inline adam-cli/src/test/scala/org/bdgenomics/adam/cli/ADAM2FastqSuite.scala]: Er, not the end of the imports, but after `com.google`.

File: adam-core/src/test/scala/org/bdgenomics/adam/converters/AlignmentRecordConverterSuite.scala [hunk 4]
```diff
@@ -17,6 +17,9 @@
  */
 package org.bdgenomics.adam.converters
 
+import java.io.File
+
```
Comment (fnothaft) [inline adam-core/src/test/scala/org/bdgenomics/adam/converters/AlignmentRecordConverterSuite.scala]: Nit: Extra space.

**Comment by fnothaft** [comment]
Other than a few nits, looks good to me! Thanks @arahuja.

Also, the failing test in Jenkins appears to be `org.bdgenomics.adam.cli.Adam2FastqSuite.convert SAM to paired FASTQ`. The stack trace is:

```
java.lang.NullPointerException
      at scala.collection.mutable.ArrayOps$ofRef$.length$extension(ArrayOps.scala:114)
      at scala.collection.mutable.ArrayOps$ofRef.length(ArrayOps.scala:114)
      at scala.collection.SeqLike$class.size(SeqLike.scala:106)
      at scala.collection.mutable.ArrayOps$ofRef.size(ArrayOps.scala:108)
      at scala.collection.mutable.Builder$class.sizeHint(Builder.scala:69)
      at scala.collection.mutable.ArrayBuilder.sizeHint(ArrayBuilder.scala:24)
      at scala.collection.TraversableLike$class.builder$1(TraversableLike.scala:240)
      at scala.collection.TraversableLike$class.map(TraversableLike.scala:243)
      at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:108)
      at org.bdgenomics.adam.rdd.ADAMContext.loadBam(ADAMContext.scala:239)
      at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loadAlignments$1.apply(ADAMContext.scala:516)
      at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loadAlignments$1.apply(ADAMContext.scala:513)
      at org.apache.spark.rdd.Timer.time(Timer.scala:57)
      at org.bdgenomics.adam.rdd.ADAMContext.loadAlignments(ADAMContext.scala:511)
      at org.bdgenomics.adam.cli.Adam2FastqSuite$$anonfun$1.apply$mcV$sp(ADAM2FastqSuite.scala:42)
      at org.bdgenomics.utils.misc.SparkFunSuite$$anonfun$sparkTest$1.apply$mcV$sp(SparkFunSuite.scala:97)
      at org.bdgenomics.utils.misc.SparkFunSuite$$anonfun$sparkTest$1.apply(SparkFunSuite.scala:93)
      at org.bdgenomics.utils.misc.SparkFunSuite$$anonfun$sparkTest$1.apply(SparkFunSuite.scala:93)
      at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
      at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
      at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
      at org.scalatest.Transformer.apply(Transformer.scala:22)
      at org.scalatest.Transformer.apply(Transformer.scala:20)
      at org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:166)
      at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
      at org.scalatest.FunSuite.withFixture(FunSuite.scala:1555)
      at org.scalatest.FunSuiteLike$class.invokeWithFixture$1(FunSuiteLike.scala:163)
      at org.scalatest.FunSuiteLike$$anonfun$runTest$1.apply(FunSuiteLike.scala:175)
      at org.scalatest.FunSuiteLike$$anonfun$runTest$1.apply(FunSuiteLike.scala:175)
      at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
      at org.scalatest.FunSuiteLike$class.runTest(FunSuiteLike.scala:175)
```

**Comment by arahuja** [comment]
Imports and license cleaned up

Test pasts for me locally, but possibly some weirdness in the paths on Jenkins.  Updated it, let's see if it works, if not is there somewhere to view the Jenkins page?

**Comment by arahuja** [comment]
Jenkins, test this please.

**Comment by arahuja** [comment]
Looks I had to move where the test resources file were, should work now.

**Comment by arahuja** [comment]
Jenkins, test this please.

**Comment by fnothaft** [comment]
Merged! Thanks @arahuja.

**Pull request merged**